### ISSUE TITLE
need 64 bit support

### ISSUE THREAD

**Issue by Gerst20051** [mpociot/titanium-mobile-zip-module#3]

**Comment by Gerst20051** [comment]
I updated the settings myself but I'm getting build failed when I run build.py. @mpociot Can you help me build this myself?

**Comment by Gerst20051** [comment]
I fixed one error by downloading this Zip file to my ~/Downloads/ZipArchive folder. https://github.com/mattconnolly/ZipArchive

**Comment by Gerst20051** [comment]
I fixed the next error by updating the TITANIUM_SDK path in file titanium.xcconfig.

I changed this path to ```~/Library/Application Support/Titanium/mobilesdk/osx/$(TITANIUM_SDK_VERSION)```

**Comment by Gerst20051** [comment]
Current error. Any ideas how to fix this?

clang: error: no such file or directory: '~/Downloads/ZipArchive/ZipArchive.mm'

**Comment by Gerst20051** [comment]
After i fixed the build phases i ran into another problem related to my earlier fix. The TITANIUM_SDK path must not be relative it must be absolute. So the path should be:

```/Users/[%user%]/Library/Application Support/Titanium/mobilesdk/osx/$(TITANIUM_SDK_VERSION)```

**Comment by Gerst20051** [comment]
I couldn't find the file ZipArchive.mm because the github version at https://github.com/mattconnolly/ZipArchive is wrong. The correct version is at https://code.google.com/p/ziparchive/downloads/list.

**Comment by Gerst20051** [comment]
GOT IT!!!

![screen shot 2015-01-20 at 2 22 50 pm](https://cloud.githubusercontent.com/assets/235372/5824079/f880e808-a0af-11e4-8b71-e4af69cc0bc9.png)

**Comment by mpociot** [comment]
Great! It's also available in the ist folder now

**Comment by Gerst20051** [comment]
@mpociot do you know why the build fails when i change the version of this module from 1.1 to 1.2.0 using titanium sdk 3.4.1GA? do i need to use sdk 3.5.0GA to upgrade to 1.2.0?

**Comment by mpociot** [comment]
Can you provide me with a log file on what fails?

**Comment by Gerst20051** [comment]
i couldn't figure it out it. wasn't adding anything to .metadata/.log file i'll just use the old version for now.

**Comment by mpociot** [comment]
Well..but can you give me the build log?
Where do you see an error? And most important *which* error? ;)

**Comment by Gerst20051** [comment]
[INFO] :   Invoking xcodebuild
[ERROR] :  ** BUILD FAILED **
[ERROR] :  The following build commands failed:
[ERROR] :  	Ld build/Debug-iphonesimulator/Passport.app/Passport normal i386
[ERROR] :  (1 failure)

**Comment by mpociot** [comment]
Okay...that's not helping at all :)
Can you open the xcodeproj file in your `build` folder and try to run it manually from xcode?

**Comment by Gerst20051** [comment]
is there a command line method to do that or do i need to open the build folder from xcode?

**Comment by Gerst20051** [comment]
@mpociot any ideas?

![screen shot 2015-01-20 at 6 26 11 pm](https://cloud.githubusercontent.com/assets/235372/5828425/f674c3f0-a0d1-11e4-9ca2-b968dd76d9d7.png)

**Comment by Gerst20051** [comment]
I've seen other people having this symbols not defined before. did something in that commit break it from working in a simulator?

**Comment by Gerst20051** [comment]
I just learned how to "Expand All Transcripts".

Undefined symbols for architecture i386:
  "_OBJC_CLASS_$_Zip", referenced from:
      objc-class-ref in libde.marcelpociot.zip.a(DeMarcelpociotZipModule.o)
     (maybe you meant: _OBJC_CLASS_$_ZipArchive)
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Comment by Gerst20051** [comment]
@mpociot any luck with this?

**Comment by giantfrog** [comment]
Same problem here

**Comment by jonasschoch** [comment]
this error:
Undefined symbols for architecture i386:
"OBJC_CLASS$Zip", referenced from:
objc-class-ref in libde.marcelpociot.zip.a(DeMarcelpociotZipModule.o)
(maybe you meant: _OBJC_CLASS$_ZipArchive)
ld: symbol(s) not found for architecture i386
clang: error: linker command failed with exit code 1 (use -v to see invocation)

It commes from the dynamic library libz.1.2.5.dylib. replace that lib with the static libz.a form here http://zlib.net

**Comment by morinel** [comment]
@jonasschoch: I see the same error you're refering to above, but could you please tell me which libz.a I should download from http://zlib.net and where I should put it?

**Comment by jonasschoch** [comment]
@morinel Just download the latest source code release, currently v1.2.8 here: http://zlib.net/zlib-1.2.8.tar.gz
extract this files and enter the directory in your terminal. There is a makefile, simply enter "make".
You will get the libz.a among other libs...

Then, open the xcode-project and somewhere in the file inspector, you will see a libz.1.2.5.dylib. replace that dynamic library with your freshly built libz.a, and clean-build the xcode-project. done.

**Comment by morinel** [comment]
Thanks Jonas, but unfortunately it doesn't work. I still see the same error while compiling for the iOS Simulator. Doesn't the command below still use the system's default libz.a / libzdylib?

```bash
[TRACE]     /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch i386 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator8.1.sdk -L/Users/USER/Projects/PROJECT/build/iphone/build/Debug-iphonesimulator -L/Users/USER/Projects/PROJECT/build/iphone/lib -L/Users/USER/Library/Application\ Support/Titanium/modules/iphone/de.marcelpociot.zip/1.2.0 -L/Users/USER/Projects/PROJECT/modules/iphone/ti.map/2.2.0 -L/Users/USER/Projects/PROJECT/modules/iphone/ti.storekit/3.1.2 -L/Users/USER/Projects/PROJECT/modules/iphone/bencoding.securely/1.53.B -L/Users/USER/Projects/PROJECT/modules/iphone/ti.charts/1.5.1 -L/Users/USER/Projects/PROJECT/build/iphone/Classes/APSAnalytics -L/Users/USER/Projects/PROJECT/build/iphone/Classes/APSHTTPClient -F/Users/USER/Projects/PROJECT/build/iphone/build/Debug-iphonesimulator -filelist /Users/USER/Projects/PROJECT/build/iphone/build/PROJECT.build/Debug-iphonesimulator/PROJECT.build/Objects-normal/i386/PROJECT.LinkFileList -dead_strip -Xlinker -objc_abi_version -Xlinker 2 -ObjC -ObjC -framework QuartzCore -ObjC -framework QuartzCore -framework Security -framework Security -framework StoreKit -framework Security -framework StoreKit -framework Security -weak_framework MapKit -weak_framework MapKit -lz -lz -weak_framework iAd -fobjc-link-runtime -Xlinker -no_implicit_dylibs -stdlib=libc++ -mios-simulator-version-min=7.1 -lti_ios_profiler -framework CoreMedia -weak_framework CoreMotion -lAPSAnalytics -weak_framework CoreText -framework EventKit -framework EventKitUI -weak_framework Security -licucore -framework Foundation -weak_framework UIKit -framework CoreGraphics -weak_framework AddressBook -framework AddressBookUI -framework CFNetwork -framework CoreLocation -weak_framework MapKit -framework MessageUI -framework MobileCoreServices -framework OpenGLES -framework QuartzCore -framework SystemConfiguration -lsqlite3 -lz -framework AudioToolbox -lti_ios_debugger -weak_framework MediaPlayer -framework AVFoundation -lxml2 -framework StoreKit -lTiCore -lde.marcelpociot.zip -lti.map -lti.storekit -lbencoding.securely -lti.charts -lAPSHTTPClient -framework ExternalAccessory -ltiverify -Xlinker -dependency_info -Xlinker /Users/USER/Projects/PROJECT/build/iphone/build/PROJECT.build/Debug-iphonesimulator/PROJECT.build/Objects-normal/i386/PROJECT_dependency_info.dat -o /Users/USER/Projects/PROJECT/build/iphone/build/PROJECT.build/Debug-iphonesimulator/PROJECT.build/Objects-normal/i386/PROJECT
```
Additionally, how do I tell while building libz 1.2.8 which architectures it should create?

**Comment by DeveloperLVC** [comment]
I am having the same problem with the module. When using 1.2.0 I gain two errors similar to @Gerst20051. I have tried with both Titanium SDK 3.5.1 and 3.5.0.

Any ideas on how to rectify this?

Thanks

**Comment by markive** [comment]
Did anyone manage to sort this? It seems there is definitely a problem with the last commit?

**Comment by morinel** [comment]
I decided to move to [ti.compression](http://gitt.io/component/ti.compression) instead. Maybe this works for you too.

**Comment by markive** [comment]
Yeah I used to use ti.compression, but it blows up the app if the zip file is really big. The second issue is there is no progress event so the app just freezes until it's unzipped.. :(

**Comment by Gerst20051** [comment]
we are also using ti.compression now.
